py3status/modules/sysdata.py

```diff
@@ -267,13 +267,37 @@ def _get_cputemp(self, zone, unit):
         m = re.search("(Core 0|CPU Temp).+\+(.+).+\(.+", sensors)
         if m:
             cpu_temp = float(m.groups()[1].strip()[:-2])
-            if unit == "K":
-                cpu_temp += 273.15
         else:
             cpu_temp = "?"
 
         return cpu_temp
 
+    def _get_cputemp(self, zone, zone_path, unit):
+
+        if unit not in [u"°F", u"°C", "K"]:
+            return "unknown unit"
+
+        if zone_path is not None:
+            try:
+                with open(zone_path) as f:
+                    cpu_temp = f.readline()
+                cpu_temp = float(cpu_temp) / 1000  # convert from mdegC to degC
+            except IOError:
+                cpu_temp = "zone_path does not exist"
+            except ValueError:
+                cpu_temp = "?"
+
+        else:
+            cpu_temp = self._get_cputemp_with_lmsensors(zone)
```

Comment:
In the comment in the top you said that "zone" has priority over "zone_path" (if "zone" is None, try to use "zone_path"), but the logic doesn't seem to match that description, "zone_path" will be used even if "zone" is not None. Can you make them consistent?